code/modules/mob/living/carbon/human/species/station/prometheans.dm

```diff
@@ -150,15 +159,31 @@ var/datum/species/shapeshifter/promethean/prometheans
 			if (istype(T, /turf/simulated))
 				var/turf/simulated/S = T
 				S.dirt = 0
-			H.nutrition += rand(15, 45)
+			H.nutrition = min(400, max(0, rand(15, 30)))
```

Comment:
This will force their nutrition to somewhere on [15, 30] whenever they eat dirt, instead of giving them 15-30 more nutrition.

Also, the if statement on line 155 enclosing this block should be
`if(C && !(H.shoes || (H.wear_suit && (H.wear_suit.bodyparts_covered & FEET))))`
The current implementation will return if they step in dirt and are wearing shoes, which prevents the regen from processing.